lib/rubocop/cop/rspec/named_subject.rb

```diff
@@ -62,14 +99,45 @@ def on_block(node)
           return if !rspec_block?(node) || ignored_shared_example?(node)
 
           subject_usage(node) do |subject_node|
-            add_offense(subject_node.loc.selector)
+            check_explicit_subject(subject_node)
           end
         end
 
+        private
+
         def ignored_shared_example?(node)
           cop_config['IgnoreSharedExamples'] &&
             node.each_ancestor(:block).any?(&method(:shared_example?))
         end
+
+        def check_explicit_subject(node)
+          return unless disallow_explicit_subject?(node)
+
+          add_offense(node.loc.selector)
+        end
+
+        def disallow_explicit_subject?(node)
```

Comment:
I personally find it hard to understand what the method does just looking at its name.
If the explicit subject is `user`, why it would be disallowed?

Reply:
Maybe call it a `allow_explicit_subject?` to avoid double negation (`unless disallow_...` -> `if allow_...`)?

Reply:
Have changed it to `allow_explicit_subject?` Let me know if this is clearer !

Reply:
Much better, thanks!